tests/function-controller/cmd/main.go

```diff
@@ -2,57 +2,70 @@ package main
 
 import (
 	"math/rand"
-	"testing"
-	"time"
-
-	"k8s.io/client-go/rest"
-
-	"github.com/sirupsen/logrus"
+	"os"
 
 	"github.com/kyma-project/kyma/tests/function-controller/pkg/step"
+	"github.com/kyma-project/kyma/tests/function-controller/testsuite"
 	"github.com/kyma-project/kyma/tests/function-controller/testsuite/scenarios"
-
-	"github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
 	"github.com/vrischmann/envconfig"
-	_ "k8s.io/client-go/plugin/pkg/client/auth"
-	controllerruntime "sigs.k8s.io/controller-runtime"
+	"k8s.io/client-go/rest"
 
-	"github.com/kyma-project/kyma/tests/function-controller/testsuite"
-)
+	"time"
 
-type config struct {
-	Test testsuite.Config
-}
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
 
-func TestRuntimes(t *testing.T) {
-	runTests(t, scenarios.FunctionTestStep)
+var availableScenarios = map[string][]testSuite{
+	"serverless-integration": {scenarios.SimpleFunctionTest},
+	"kyma-integration":       {scenarios.FunctionTestStep, scenarios.GitopsSteps},
 }
 
-func TestGitSourcesFunctions(t *testing.T) {
-	runTests(t, scenarios.GitopsSteps)
+type config struct {
+	Test testsuite.Config
 }
 
-type testSuite func(*rest.Config, testsuite.Config, *logrus.Entry) (step.Step, error)
+func main() {
+	logf := logrus.New()
+	logf.SetFormatter(&logrus.TextFormatter{})
+	logf.SetReportCaller(false)
 
-func runTests(t *testing.T, testFunc testSuite) {
-	rand.Seed(time.Now().UnixNano())
-	g := gomega.NewGomegaWithT(t)
+	if len(os.Args) < 2 {
+		logf.Errorf("Scenario not specified. Specify it as the first argument")
+		os.Exit(2)
+	}
```

Comment:
it looks kinda clunky, especially given that flags are listed in os.Args, go back to flags or envs
see:
```go
package main

import (
	"flag"
	"fmt"
	"os"
)

func main() {
	data := flag.Bool("test", false, "sample usage")
	flag.Parse()

	for i, item := range os.Args {
		fmt.Printf("Os arg #%d: %s\n", i+1, item)
	}

	fmt.Println(*data)
}
![image](https://user-images.githubusercontent.com/17271979/100081965-ed140500-2e47-11eb-83a9-b17a267c0760.png)

```

Reply:
discussed offline - it stays